### PR TITLE
Support for OpenTelemetry exporter

### PR DESCRIPTION
**Pull request by josunect**
Fixes https://github.com/kiali/kiali/issues/5569

### Configuration

Maybe it is a bit confusing, but, the idea is to just set up or the `jaeger_collector` OR the `otel_collector`
- The Jaeger collector by default sends traces to `http://0.0.0.0:14268/api/traces`. It needs the protocol in the URL. 
- The OTel collector can use http, https or grpc. There is a TODO to improve grpc config using TLS (Need to set up the cert parameters). The URL does not use the protocol. By default, the http exporter sends the traces to `/v1/traces`

Configuration example: 

```
  observability: 
    tracing: 
      collector_url: "localhost:4318"
      enabled: true
      otel_collector: 
        enabled: true
        protocol: "http"
      jaeger_collector: 
        enabled: false
```

### Compatibility

Not sure how this configuration could work with backwards compatibility, as now the jaeger_collector would need to be enabled (If not, it will throw an error)

### Test

The OTel collectors were tested using Grafana Tempo and Tanka. 
- Visit the workloads list page
- curl (It should return the traces generated by Kiali): `curl -G -s http://localhost:16686/api/search --data-urlencode 'tags=package=business'`

### Operator 

Operator PR: https://github.com/kiali/kiali-operator/pull/673

### CODE REVIEW

**Comment by jmazzitelli** [comment]
You should do the same kind of thing we do in the `auth` section (see https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.auth). We have "auth.strategy" and whatever value you set in there, there is an optional section with the same name that contains configuration specific for that strategy.

For example:

```
auth:
  strategy: openshift
  openshift:
    client_id_prefix: foo
    token_inactivity_timeout: 5
    ...and others...
```

But if auth.strategy is "openid", then there is a separate section for that:

```
auth:
  strategy: openid
  openid:
    client_id: foo
    disable_rbac: true
    ... and others...
```

So, what this means is, for your new setting, you would add a new `collector_impl` or `collector_type` setting (whatever name you think is best), and each type would have its own separate section like you have (but remove each "enabled" flag that you added - no need for that since this new "collector_impl" or "collector_type" setting would tell you which is enabled)... like this:

```
observability: 
    tracing: 
      collector_url: "localhost:4318"
      enabled: true
      collector_type: "otel" # each value ("otel" or "jaeger") would have a matching section below.
      otel: 
        protocol: "http"
      jaeger: 
        something_here: foo
```

Note that if, say, jaeger doesn't need any extra config, then you don't have to worry about its own section (just like for "auth.strategy: anonymous" we do not have a "auth.anonymous" section because there is nothing extra to configure for anonymous mode).

**Comment by josunect** [comment]
> You should do the same kind of thing we do in the `auth` section (see https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.auth). We have "auth.strategy" and whatever value you set in there, there is an optional section with the same name that contains configuration specific for that strategy.
> 
> For example:
> 
> ```
> auth:
>   strategy: openshift
>   openshift:
>     client_id_prefix: foo
>     token_inactivity_timeout: 5
>     ...and others...
> ```
> 
> But if auth.strategy is "openid", then there is a separate section for that:
> 
> ```
> auth:
>   strategy: openid
>   openid:
>     client_id: foo
>     disable_rbac: true
>     ... and others...
> ```
> 
> So, what this means is, for your new setting, you would add a new `collector_impl` or `collector_type` setting (whatever name you think is best), and each type would have its own separate section like you have (but remove each "enabled" flag that you added - no need for that since this new "collector_impl" or "collector_type" setting would tell you which is enabled)... like this:
> 
> ```
> observability: 
>     tracing: 
>       collector_url: "localhost:4318"
>       enabled: true
>       collector_type: "otel" # each value ("otel" or "jaeger") would have a matching section below.
>       otel: 
>         protocol: "http"
>       jaeger: 
>         something_here: foo
> ```
> 
> Note that if, say, jaeger doesn't need any extra config, then you don't have to worry about its own section (just like for "auth.strategy: anonymous" we do not have a "auth.anonymous" section because there is nothing extra to configure for anonymous mode).

Thank you, @jmazzitelli ! This is very useful.

**Review by aljesusg — APPROVED** [review]
LGFM code no tested, I'l try it later

File: observability/tracing.go [hunk 1]
```diff
@@ -117,3 +134,72 @@ func StartSpan(ctx context.Context, funcName string, attrs ...attribute.KeyValue
...
+
+					// TODO: Support TLS
+					exporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure(),
+						otlptracegrpc.WithEndpoint(collectorURL),
+						otlptracegrpc.WithDialOption(grpc.WithBlock()),
+					)
+
+					return exporter, err
+				} else {
+					return nil, errors.New("Error in configuration options getting the observability exporter")
```
Comment (jmazzitelli) [inline observability/tracing.go:197]: Actually, make these error messages more specified. This error and the one on line 201 are identical - when you see this, you have no idea if its the protocol that is bad or the collector type that is bad.
```suggestion
					return nil, errors.New("Error in configuration options getting the observability exporter - invalid otel.protocol.")
```

File: observability/tracing.go [hunk 2]
```diff
@@ -117,3 +134,72 @@ func StartSpan(ctx context.Context, funcName string, attrs ...attribute.KeyValue
...
+				}
+			}
+		} else {
+			return nil, errors.New("Error in configuration options getting the observability exporter")
```
Comment (jmazzitelli) [inline observability/tracing.go:201]:
```suggestion
			return nil, errors.New("Error in configuration options getting the observability exporter. Invalid collector type.")
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
a couple minor changes needed

**Review by jmazzitelli — DISMISSED** [review]
lgtm

**Review by jmazzitelli — APPROVED** [review]

**Pull request merged**